src/embed/dc-resize.js

```diff
@@ -0,0 +1,21 @@
+import { EMBED_URL } from "@/config/embed.js";
+
+window.addEventListener("message", ({ origin, data }) => {
+  // normalize origins for checking
+  if (new URL(origin).origin !== new URL(EMBED_URL).origin) return;
+
+  // ensure all the data we need
+  if (![data.width, data.height, data.href].every(Boolean)) return;
+
+  // find all the iframe with this href
+  const iframes = document.querySelectorAll(`iframe[src*="${data.href}"]`);
```

Comment:
This contains selector could lead to bugs through unlikely, but possible partial matches (e.g. `/document/123` would also match `/document/1234`). A more robust solution would be URL comparison:
```
const iframes = Array.from(document.querySelectorAll(`iframe`)).filter(iframe => {
  try {
    const iframeSrc = new URL(iframe.src);
    const dataSrc = new URL(data.href);
    return iframeSrc.path === dataSrc.path;
    // could go further with search string comparison
    // && iframeSrc.search === dataSrc.search;
  } catch {
    return iframe.src === data.href;
  }
})
```